zoo/repos/utils.py

```diff
@@ -84,16 +84,24 @@ def _parse_file(path, base=None):
 def openapi_definition(repository, request=None, repo_path=None):
     redis_conn = redis.get_connection(decode_responses=True)
     specs = []
+    repo_dir = None
 
     if repo_path is None:
-        with tempfile.TemporaryDirectory() as repo_dir:
-            try:
-                repo_path = download_repository(repository, repo_dir)
-            except (MissingFilesError, RepositoryNotFoundError) as err:
-                log.info("repos.views.openapi.git_error", repo=repository, error=err)
-                return JsonResponse(
-                    {"error": "error downloading repository"}, status=500
-                )
+        try:
+            repo_dir = tempfile.TemporaryDirectory()
+        except (FileExistsError, PermissionError) as err:
+            log.info("repos.views.tmpdir", repo=repository, error=err)
+            return JsonResponse(
```

Comment:
not related to your changes, but this isn't correct
the function should only return a list, the responses will be handled on the views

so... change both exception handlings to return empty list